spec/filters/http_spec.rb

```diff
@@ -1,39 +1,75 @@
 require 'logstash/devutils/rspec/spec_helper'
+require 'logstash/plugin_mixins/ecs_compatibility_support/spec_helper'
 require 'logstash/filters/http'
 
 describe LogStash::Filters::Http do
   subject { described_class.new(config) }
   let(:event) { LogStash::Event.new(data) }
   let(:data) { { "message" => "test" } }
 
-  describe 'response body handling' do
-    before(:each) { subject.register }
+  describe 'response body handling', :ecs_compatibility_support do
+
     let(:url) { 'http://laceholder.typicode.com/users/10' }
-    let(:config) do
-      { "url" => url, "target_body" => 'rest' }
-    end
+    let(:config) { { "url" => url } }
+
     before(:each) do
+      subject.register
+
       allow(subject).to receive(:request_http).and_return(response)
       subject.filter(event)
     end
 
-    context "when body is text" do
+    ecs_compatibility_matrix(:disabled, :v1) do |ecs_select|
+
+      let(:config) { super().merge "ecs_compatibility" => ecs_compatibility }
+
       let(:response) { [200, {}, "Bom dia"] }
 
-      it "fetches and writes body to target" do
-        expect(event.get('rest')).to eq("Bom dia")
+      context "when body is text" do
+
+        it "fetches and writes body to target" do
+          if ecs_select.active_mode == :disabled
+            expect(event.get('body')).to eq("Bom dia")
+          else
+            expect(event.get('[http][request][body][content]')).to eql 'Bom dia'
+          end
+        end
+
       end
-    end
-    context "when body is JSON" do
-      context "and headers are set correctly" do
-        let(:response) { [200, {"content-type" => "application/json"}, "{\"id\": 10}"] }
+
+      context "when body is JSON" do
+        context "and headers are set correctly" do
+
+          let(:response) { [200, {"content-type" => "application/json"}, "{\"id\": 10}"] }
+
+          it "fetches and writes body to target" do
+            if ecs_select.active_mode == :disabled
+              expect(event.get('[body][id]')).to eq(10)
+            else
+              expect(event.include?('[body]')).to be false
+
+              # TODO: this is going to cause issues with an ECS template!?
```

Comment:
what do we do here, do we introduce a `parse_json => true|false` configuration option?
the option would be on in legacy mode and `false` in ECS mode ... :yin_yang: